### PR TITLE
3111 - Redirects don't happen after workflow deletion

### LINKED ISSUE #2762: [bug]: Exception Thrown When Deleting Workflow
### Describe the bug

An error occurs when attempting to delete a workflow. The application returns an error message stating "Not Found" instead of successfully deleting the workflow.

### Steps To Reproduce

https://github.com/user-attachments/assets/11ef7c3c-5cfa-43b4-b533-6ed792a0946f

### Expected behavior

-

### Version

latest

### OS

ubuntu

### Browser

_No response_

### Additional context

_No response_

### LINKED ISSUE #3111: [bug]: Redirects don't happen after workflow deletion
### Describe the bug

When I delete the workflow directly from the settings, I am expecting a redirect to happen either to the previous page or to the workflow list. But instead, the URL stays the same. (In the video, I manually triggered the refresh.)

https://github.com/user-attachments/assets/fa0fa30b-e096-4f74-b9d7-21c4b772d19c

### Steps To Reproduce

1. Go to https://app.bytechef.io/automation/projects and open any workflow.
2. Go to the workflow settings and delete it.

### Expected behavior

Go to the previous page, or the list page where the workflow was stored.

### Version

Production (https://app.bytechef.io)

### OS

not sure...

### Browser

Google Chrome version 140.0.7339.80

### Additional context

Could I be assigned this issue, and what should be the behavior after workflow deletion?

### PR DESCRIPTION
**Pull request by karlocehulic19**
## Description

Fixes unnecessary refetch of workflow after deletion in the settings menu by removing the second useProject hook call. It also handles deleting the last workflow project edge case.

Fixes #3111
Fixes #2762 

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Manually test with React Query devtools

## Checklist:

- [ + ] My code follows the style guidelines of this project
- [ + ] I have performed a self-review of my own code
- [ + ] I have commented my code, particularly in hard-to-understand areas
- [ + ] I have made corresponding changes to the documentation
- [ + ] My changes generate no new warnings
- [ + ] I have added tests that prove my fix is effective or that my feature works (there were no prior tests, if needed, I could add tests
- [ + ] New and existing unit tests pass locally with my changes

I am sending the master because development has been deleted

### CODE REVIEW

**Comment by karlocehulic19** [comment]
@kresimir-coko, could you review this PR? I am not sure why CI build check fails. I ran all Client Development tasks.

**Comment by kresimir-coko** [comment]
@karlocehulic19 with your changes we get a duplicate request to the project workflows endpoint, the deletion doesn't trigger the bad request anymore which is good, but we don't want the duplicate request either.

Here's an example of the duplicated request: `http://127.0.0.1:5173/api/automation/internal/projects/1051/workflows` 

image

**Comment by karlocehulic19** [comment]
@kresimir-coko, I belive I fixed the issue now, here is the video of network requests...
[Screencast from 07.10.2025 15:17:59.webm](https://github.com/user-attachments/assets/cb3d1b53-1424-44a4-a807-b80890e1dbe8)

File: client/src/pages/automation/project/components/project-header/components/settings-menu/hooks/useSettingsMenu.ts [hunk 1]
```diff
@@ -117,7 +116,17 @@ export const useSettingsMenu = ({project, workflow}: {project: Project; workflow
                 queryKey: ProjectWorkflowKeys.projectWorkflows(projectId),
             });
 
-            navigate(`/automation/projects/${projectId}/project-workflows/${firstRemainingWorkflowId}?${searchParams}`);
+            // exact here prevents double fetchs of project
```
Comment (kresimir-coko) [inline client/src/pages/automation/project/components/project-header/components/settings-menu/hooks/useSettingsMenu.ts]: remove comment

File: client/src/pages/automation/project/components/project-header/components/settings-menu/hooks/useSettingsMenu.ts [hunk 2]
```diff
@@ -117,7 +116,17 @@ export const useSettingsMenu = ({project, workflow}: {project: Project; workflow
...
+            queryClient.invalidateQueries({
+                exact: true,
+                queryKey: ProjectKeys.project(projectId),
+            });
+
+            const baseUrl = '/automation/projects';
+            const firstRemainingWorkflowUrlSuffix = firstRemainingWorkflowId
+                ? `/${projectId}/project-workflows/${firstRemainingWorkflowId}?${searchParams}`
+                : '';
+            navigate(baseUrl + firstRemainingWorkflowUrlSuffix);
```
Comment (kresimir-coko) [inline client/src/pages/automation/project/components/project-header/components/settings-menu/hooks/useSettingsMenu.ts:131]: we want an empty line between variable declaration and its usage.

i would also like it if you rewrote the firstRemainingWorkflowUrlSuffix so that it's an empty string by default, and if firstRemainingWorkflowId exists, change its value to the complicated string...

the pattern is something like this:
```jsx
let firstRemainingWorkflowUrlSuffix = '';

if (foo) {
  firstRemainingWorkflowUrlSuffix = ...
}
```

otherwise, great job!

**Comment by kresimir-coko** [comment]
We have a weird CI issue now @karlocehulic19, can you please run `npm run test` from the `client` folder and report back.

**Comment by karlocehulic19** [comment]
> We have a weird CI issue now @karlocehulic19, can you please run `npm run test` from the `client` folder and report back.

For me, all tests pass, but here is the whole log:
npm run test

> bytechef-client@0.1.0 test
> vitest run


 RUN  v3.2.4 /home/karlo19/repos/bitna-prednost/bytechef/client

stdout | src/pages/automation/project/components/project-header/tests/LoaderNotification.test.tsx
<body />

stdout | src/pages/account/public/tests/Register.test.tsx
<body />

stdout | src/pages/account/public/tests/Login.test.tsx
<body />

 ✓ src/components/MultiSelect/MultiSelect.test.tsx (15 tests) 953ms
 ✓ src/pages/platform/workflow-editor/tests/ActionComponentsFilter.test.tsx (9 tests) 1184ms
 ✓ src/pages/home/tests/ModeSelectionDialog.test.tsx (13 tests) 1190ms
 ✓ src/pages/account/public/tests/Register.test.tsx (5 tests) 1047ms
   ✓ should set type as password initially and toggle between types when "show password" icon is clicked  422ms
 ✓ src/pages/automation/project/components/project-header/tests/LoaderNotification.test.tsx (3 tests) 2541ms
   ✓ should render the success message when it is not fetching and isOnline is true   974ms
   ✓ should render the fetching message when something is fetching and isOnline is true   799ms
   ✓ should render the offline message when isOnline is false   765ms
 ✓ src/pages/account/public/tests/Login.test.tsx (10 tests) 2102ms
   ✓ should set type as password initially and toggle between types when "show password" icon is clicked  398ms
   ✓ should not submit form when only "show password" button is clicked  426ms
   ✓ should submit form with correct login details when "Log in" is clicked  368ms
   ✓ should disable submit button and show loading icon while log in credentials are being authenticated  338ms
stdout | src/pages/automation/project/components/project-header/tests/SettingsMenu.test.tsx
<body />

stderr | src/pages/automation/project/components/project-header/tests/SettingsMenu.test.tsx > should close the dropdown on click of a button inside the Workflow tab
Error: Not implemented: navigation (except hash changes)
    at module.exports (/home/karlo19/repos/bitna-prednost/bytechef/client/node_modules/jsdom/lib/jsdom/browser/not-implemented.js:9:17)
    at navigateFetch (/home/karlo19/repos/bitna-prednost/bytechef/client/node_modules/jsdom/lib/jsdom/living/window/navigation.js:77:3)
    at exports.navigate (/home/karlo19/repos/bitna-prednost/bytechef/client/node_modules/jsdom/lib/jsdom/living/window/navigation.js:55:3)
    at LocationImpl._locationObjectNavigate (/home/karlo19/repos/bitna-prednost/bytechef/client/node_modules/jsdom/lib/jsdom/living/window/Location-impl.js:30:5)
    at LocationImpl._locationObjectSetterNavigate (/home/karlo19/repos/bitna-prednost/bytechef/client/node_modules/jsdom/lib/jsdom/living/window/Location-impl.js:24:17)
    at LocationImpl.set href [as href] (/home/karlo19/repos/bitna-prednost/bytechef/client/node_modules/jsdom/lib/jsdom/living/window/Location-impl.js:46:10)
    at Location.set href [as href] (/home/karlo19/repos/bitna-prednost/bytechef/client/node_modules/jsdom/lib/jsdom/living/generated/Location.js:125:37)
    at onClick (/home/karlo19/repos/bitna-prednost/bytechef/client/src/pages/automation/project/components/project-header/components/settings-menu/components/WorkflowTabButtons.tsx:67:49)
    at executeDispatch (/home/karlo19/repos/bitna-prednost/bytechef/client/node_modules/react-dom/cjs/react-dom-client.development.js:16368:9)
    at runWithFiberInDEV (/home/karlo19/repos/bitna-prednost/bytechef/client/node_modules/react-dom/cjs/react-dom-client.development.js:1522:13) undefined

 ✓ src/components/tests/SelectGeneric.test.tsx (7 tests) 845ms
stderr | src/pages/automation/project/components/projects-sidebar/tests/WorkflowsListFilter.test.tsx > should call setSortBy with correct value when a sort option is selected
An update to PopperContent inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
  /* fire events that update state */
});
/* assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act
An update to FocusScope inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
  /* fire events that update state */
});
/* assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act
An update to DismissableLayer inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
  /* fire events that update state */
});
/* assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act
An update to Presence inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
  /* fire events that update state */
});
/* assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act
An update to Presence inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
  /* fire events that update state */
});
/* assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act
An update to Menu inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
  /* fire events that update state */
});
/* assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act
An update to PopperContent inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
  /* fire events that update state */
});
/* assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act
An update to FocusScope inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
  /* fire events that update state */
});
/* assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act
An update to DismissableLayer inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
  /* fire events that update state */
});
/* assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act
An update to Presence inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
  /* fire events that update state */
});
/* assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act
An update to Presence inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
  /* fire events that update state */
});
/* assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act
An update to Menu inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
  /* fire events that update state */
});
/* assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act
An update to PopperContent inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
  /* fire events that update state */
});
/* assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act

 ✓ src/pages/automation/project/components/projects-sidebar/tests/WorkflowsListFilter.test.tsx (4 tests) 605ms
 ✓ src/pages/automation/project/components/project-header/tests/SettingsMenu.test.tsx (4 tests) 1075ms
   ✓ should open the settings dropdown menu on click of a settings button  411ms
 ✓ src/pages/account/public/tests/PasswordResetInit.test.tsx (5 tests) 625ms
stdout | src/pages/automation/project/components/project-header/tests/WorkflowSelect.test.tsx
<body />

 ✓ src/pages/automation/project/components/project-header/tests/WorkflowSelect.test.tsx (4 tests) 529ms
 ✓ src/pages/automation/project/components/project-header/components/settings-menu/components/ProjectTabButtons/ProjectTabButtons.test.tsx (7 tests) 509ms
stdout | src/pages/automation/project/components/project-header/tests/PublishPopover.test.tsx
<body />

 ✓ src/pages/automation/project/components/projects-sidebar/ProjectsLeftSidebar.test.tsx (5 tests) 303ms
stdout | src/pages/platform/workflow-editor/components/properties/components/property-input/PropertyInput.test.tsx > PropertyInput > should change input value
Open this URL in your browser

https://testing-playground.com/#markup=DwEwlgbgfMBmYFMA2IDOCAuACAxkghqqgLwBEA7gLSwCuSSWqADvjgpQJ6UCMpM4EXASJlYSBAA8sVWvSxgMCALapKbAHaKATlgBWNVBjCwuAI0zkECdX1CQhhEqTGT5ilWuvbbBcwzyOZIoSGJSoSliwAPaalEoI4DQRTFZalOCo+KbiIABcODRaqFFp6lGh+PRRliBYKQhpGVk5uVEsOApcAOwADFji+ODqAOaUZVpKlaSRJWTK+GBIfACik4tYAIIgIFoIRMAA9L7IMAcC-PYBIs7iUgrKqhrep+eHrwIO10qhvBeCV05ogVULktGARpQAEwzYGg8GjbI0dgAVh6fSBBkoEDAqDA2QQrRoGCQ4PYZXUCFs4KYRM+gNuWAAFpQAJzSah0BhaKI0dQgBJxWqmEr8nTCrSiyjU2mmUYYLT4dTMfC7TR1CSUADMdS43CwwVCpkICEYjMG1TCEXlitxRhiaiiSBKqEiiwJ4sl6LduVllGtSpYquw8HEuQNltdoeisXiiQiIYJ4eiu2G3N5tSYBDYjMdorDklCSiJAuTCFTPL5WCa+LyBSKJTG5UolSdNSrOOaCVa7U6lFRWCUeXD4SwPOJpMbFJhmOxuPxhPHFMnJoxqlneNDYIh0NXWJxG4JW4RSCRfbR0zAIDmayWdSzCBzKAaZFWC1vhgQTDIvCwKrA+EoY4kBfG9Nm2XYiGmDAOBSa832mdR8HiODFlsM5IBeDDDngZA0EwTDoCAA

stderr | src/pages/platform/workflow-editor/components/properties/components/property-input/PropertyInput.test.tsx > PropertyInput > should change input value
A component is changing an uncontrolled input to be controlled. This is likely caused by the value changing from undefined to a defined value, which should not happen. Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://react.dev/link/controlled-components

 ✓ src/pages/platform/workflow-editor/components/properties/components/property-input/PropertyInput.test.tsx (4 tests) 401ms
 ✓ src/pages/automation/project/components/project-header/tests/PublishPopover.test.tsx (3 tests) 396ms
 ✓ src/pages/automation/projects/Projects.test.tsx (4 tests) 695ms
   ✓ Projects Import Functionality > should show import dropdown menu items  441ms
 ✓ src/pages/automation/project/components/projects-sidebar/tests/WorkflowsListItem.test.tsx (5 tests) 219ms
stdout | src/pages/automation/project/components/project-header/tests/WorkflowActionsButton.test.tsx
<body />

 ✓ src/pages/automation/project/components/project-header/tests/WorkflowActionsButton.test.tsx (6 tests) 186ms
stderr | src/pages/automation/project/components/projects-sidebar/tests/ProjectWorkflowsList.test.tsx > should render project name and workflows list
Each child in a list should have a unique "key" prop.

Check the render method of `ProjectWorkflowsList`. See https://react.dev/link/warning-keys for more information.

 ✓ src/pages/automation/project/components/projects-sidebar/tests/ProjectWorkflowsList.test.tsx (3 tests) 158ms
 ✓ src/pages/platform/workflow-editor/tests/WorkflowNodeDetailsPanel.test.tsx (9 tests) 19ms
 ✓ src/components/ComboBox/ComboBox.test.tsx (1 test) 89ms
 ✓ src/pages/platform/workflow-editor/tests/encodingUtils.test.ts (16 tests) 17ms
 ✓ src/pages/platform/workflow-editor/components/properties/components/property-mentions-input/PropertyMentionsInput.test.tsx (1 test) 79ms
 ✓ src/pages/automation/project/components/project-header/tests/ProjectTitle.test.tsx (1 test) 87ms
 ✓ src/pages/platform/workflow-editor/utils/layoutUtils.test.tsx (8 tests) 12ms
 ✓ src/pages/platform/workflow-editor/tests/getTaskDispatcherContext.test.ts (15 tests) 12ms
 ✓ src/pages/platform/workflow-editor/tests/createConditionEdges.test.ts (7 tests) 6ms
 ✓ src/routes.loader.test.ts (2 tests) 5ms

 Test Files  28 passed (28)
      Tests  176 passed (176)
   Start at  10:31:25
   Duration  13.78s (transform 6.67s, setup 3.12s, collect 27.15s, tests 15.89s, environment 18.93s, prepare 5.37s)
